### PR TITLE
Split cut_plotter into view and presenter

### PR DESCRIPTION
**Pull request by MatthewBowles**
- Use some methods from `cut_presenter` and `matplotlib_cut_plotter` to create `cut_plotter_presenter`.
- Rename remaining `cut_presenter` to `cut_widget_presenter`.

**To test:**
 Ideally check all aspects of cuts and plotting cuts still function correctly.

_No associated issue_

### CODE REVIEW

File: mslice/models/cut/cut_cache.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+
+class CutCache(object):
```
Comment (martyngigg) [inline mslice/models/cut/cut_cache.py]: Good call to create a type for this. How about just `Cut` for the type name? I would naively expect a cache to generally be a list of old cuts - this essentially encapsulates a cut as far as I can see?

Comment (MatthewBowles) [reply mslice/models/cut/cut_cache.py]: I agree. I'll change the slice equivalent as well.

**Comment by MatthewBowles** [comment]
Thanks Martyn, should be fixed now. 

I'm aware of the remaining problem with the legends control as @aybamidele noticed it earlier on master  - that will be solved in another PR.

**Comment by martyngigg** [comment]
Thanks, Matthew. The legend thing is minor anyway as the toolbar button still works.

**Pull request merged**